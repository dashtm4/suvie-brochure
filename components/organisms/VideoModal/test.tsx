import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { VideoModal } from '.'

describe('<VideoModal />', () => {
  const children = 'test'
  const toggleModal = jest.fn()

  it('should renders correct <VideoModal />', () => {
    const wrapper = shallow(
      <VideoModal toggleModal={toggleModal}>{children}</VideoModal>
    )

    expect(wrapper.contains(children)).toBeTruthy()
    expect(wrapper.find('.aspect-w-16.aspect-h-9')).toHaveLength(0)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has correct video attributes from props', () => {
    const videoSrc = 'video.mp4'
    const posterSrc = 'image.jpg'
    const videoAutoPlay = false
    const videoControls = true
    const videoLoop = false
    const videoMuted = false

    const wrapper = shallow(
      <VideoModal
        toggleModal={toggleModal}
        videoSrc={videoSrc}
        posterSrc={posterSrc}
        videoAutoPlay={videoAutoPlay}
        videoControls={videoControls}
        videoLoop={videoLoop}
        videoMuted={videoMuted}
      >
        {children}
      </VideoModal>
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should not has wide screen aspect ratio when isWidescreenAspectRatio prop is false', () => {
    const isWidescreenAspectRatio = false

    const wrapper = shallow(
      <VideoModal
        toggleModal={toggleModal}
        isWidescreenAspectRatio={isWidescreenAspectRatio}
      >
        {children}
      </VideoModal>
    )

    expect(wrapper.find('.aspect-w-16.aspect-h-9')).toHaveLength(0)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has wide screen aspect ratio when isWidescreenAspectRatio prop is true', () => {
    const isWidescreenAspectRatio = true

    const wrapper = shallow(
      <VideoModal
        toggleModal={toggleModal}
        isWidescreenAspectRatio={isWidescreenAspectRatio}
      >
        {children}
      </VideoModal>
    )

    expect(wrapper.find('.aspect-w-16.aspect-h-9')).toHaveLength(1)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
