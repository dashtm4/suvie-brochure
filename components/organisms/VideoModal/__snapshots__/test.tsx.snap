// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoModal /> should has correct video attributes from props 1`] = `
<Modal
  toggleModal={[MockFunction]}
>
  <div
    className="md:flex"
    style={
      Object {
        "minHeight": "none",
      }
    }
  >
    <div
      className="md:w-1/2 md:order-1"
    >
      <figure
        className="relative w-full overflow-hidden h-60 bg-secondary md:h-full"
        role="none"
      >
        <VideoPlayer
          autoplay={false}
          className="absolute inset-0 object-cover w-full h-full"
          controls={true}
          loop={false}
          muted={false}
          playsInline={true}
          poster="image.jpg"
          style={
            Object {
              "height": "calc(100% + 2px)",
            }
          }
          url="video.mp4"
        />
      </figure>
    </div>
    <div
      className="px-3 py-8 md:w-1/2 md:py-12 md:pl-8 md:pr-12 md:order-0 md:flex md:flex-col md:justify-center"
    >
      test
    </div>
  </div>
</Modal>
`;

exports[`<VideoModal /> should has wide screen aspect ratio when isWidescreenAspectRatio prop is true 1`] = `
<Modal
  toggleModal={[MockFunction]}
>
  <div
    className="md:flex aspect-w-16 aspect-h-9"
    style={
      Object {
        "minHeight": "none",
      }
    }
  >
    <div
      className="md:w-1/2 md:order-1"
    >
      <figure
        className="relative w-full overflow-hidden h-60 bg-secondary md:h-full"
        role="none"
      >
        <VideoPlayer
          autoplay={true}
          className="absolute inset-0 object-cover w-full h-full"
          controls={false}
          loop={true}
          muted={true}
          playsInline={true}
          style={
            Object {
              "height": "calc(100% + 2px)",
            }
          }
        />
      </figure>
    </div>
    <div
      className="px-3 py-8 md:w-1/2 md:py-12 md:pl-8 md:pr-12 md:order-0 md:flex md:flex-col md:justify-center"
    >
      test
    </div>
  </div>
</Modal>
`;

exports[`<VideoModal /> should not has wide screen aspect ratio when isWidescreenAspectRatio prop is false 1`] = `
<Modal
  toggleModal={[MockFunction]}
>
  <div
    className="md:flex"
    style={
      Object {
        "minHeight": "none",
      }
    }
  >
    <div
      className="md:w-1/2 md:order-1"
    >
      <figure
        className="relative w-full overflow-hidden h-60 bg-secondary md:h-full"
        role="none"
      >
        <VideoPlayer
          autoplay={true}
          className="absolute inset-0 object-cover w-full h-full"
          controls={false}
          loop={true}
          muted={true}
          playsInline={true}
          style={
            Object {
              "height": "calc(100% + 2px)",
            }
          }
        />
      </figure>
    </div>
    <div
      className="px-3 py-8 md:w-1/2 md:py-12 md:pl-8 md:pr-12 md:order-0 md:flex md:flex-col md:justify-center"
    >
      test
    </div>
  </div>
</Modal>
`;

exports[`<VideoModal /> should renders correct <VideoModal /> 1`] = `
<Modal
  toggleModal={[MockFunction]}
>
  <div
    className="md:flex"
    style={
      Object {
        "minHeight": "none",
      }
    }
  >
    <div
      className="md:w-1/2 md:order-1"
    >
      <figure
        className="relative w-full overflow-hidden h-60 bg-secondary md:h-full"
        role="none"
      >
        <VideoPlayer
          autoplay={true}
          className="absolute inset-0 object-cover w-full h-full"
          controls={false}
          loop={true}
          muted={true}
          playsInline={true}
          style={
            Object {
              "height": "calc(100% + 2px)",
            }
          }
        />
      </figure>
    </div>
    <div
      className="px-3 py-8 md:w-1/2 md:py-12 md:pl-8 md:pr-12 md:order-0 md:flex md:flex-col md:justify-center"
    >
      test
    </div>
  </div>
</Modal>
`;
